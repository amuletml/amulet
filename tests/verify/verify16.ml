external val print : int -> unit = "print"

let id x = x

let main =
  let rec f = id (fun _ -> g) 1
  and g = id (fun _ -> f) 1
  print f
