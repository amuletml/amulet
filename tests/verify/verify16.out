verify16.ml[6:7 ..6:7]: error
  Invalid recursive right-hand side for variable f
  • Note: because evaluation of the variable g is not delayed
  Arising from use of the binding
  │ 
6 │   let f = id (fun _ -> g) 1
  │ 
verify16.ml[7:7 ..7:7]: error
  Invalid recursive right-hand side for variable g
  • Note: because evaluation of the variable f is not delayed
  Arising from use of the binding
  │ 
7 │   and g = id (fun _ -> f) 1
  │ 
